cmd/controller/controller.go

```diff
@@ -283,30 +281,92 @@ func getReleaseName(r *helmCrdV1.HelmRelease) string {
 	return rname
 }
 
+func findIndex(target string, s []string) int {
+	res := -1
+	for i := range s {
+		if s[i] == target {
+			res = i
+		}
+	}
+	return res
+}
+func pull(item string, s []string) ([]string, error) {
+	index := findIndex(item, s)
+	if index == -1 {
+		return []string{}, fmt.Errorf("%s not present in %v", item, s)
+	}
+	return append(s[:index], s[index+1:]...), nil
```

Comment:
Fwiw, if we don't care about the order, then it's faster to swap another element into the "gap" and reslice:

```go
func removeIndex(i int, s []string) []string {
    lastIdx = len(s) - 1
    if i != lastIdx {
        s[i] = s[lastIdx]
    }
    s[lastIdx] = ""  // drop reference to string contents
    return s[:lastIdx]
}
```